apps/user/src/user/user.service.ts

```diff
@@ -17,26 +19,99 @@ export class UserService {
     private projectRepository: Repository<ProjectCategory>,
     @InjectRepository(CareerCategory)
     private careerRepository: Repository<CareerCategory>,
+    @InjectRepository(TempUser)
+    private tempUserRepository: Repository<TempUser>,
   ) {}
 
   @Transactional()
   public async createUser(createUserDto: CreateUserDto): Promise<User> {
-    const positionCategory = await this.positionRepository.findOne({
-      where: { position_category_id: createUserDto.position_category_id },
-    });
-    const projectCategory = await this.projectRepository.findOne({
-      where: { project_category_id: createUserDto.project_category_id },
-    });
-    const careerCategory = await this.careerRepository.findOne({
-      where: { career_category_id: createUserDto.career_category_id },
-    });
+    let positionCategory, careerCategory;
+
+    if (createUserDto.position_category_id !== null && createUserDto.position_category_id !== 0) {
+      positionCategory = await this.positionRepository.findOne({
+        where: { position_category_id: createUserDto.position_category_id },
+      });
+    }
+
+    if (createUserDto.career_category_id !== null && createUserDto.career_category_id !== 0) {
+      careerCategory = await this.careerRepository.findOne({
+        where: { career_category_id: createUserDto.career_category_id },
+      });
+    }
 
     const newUser = this.userRepository.create({
       ...createUserDto,
-      position_category_id: positionCategory,
-      project_category_id: projectCategory,
-      career_category_id: careerCategory,
+      position_category_id: positionCategory?.position_category_id ?? null,
+      career_category_id: careerCategory?.career_category_id ?? null,
+      project_category_id:
+        createUserDto.project_category_id.length > 0 ? [...createUserDto.project_category_id] : [],
+      stack_category_id:
+        createUserDto.stack_category_id.length > 0 ? [...createUserDto.stack_category_id] : [],
+    });
+
+    const deleteTempUser = await this.deleteTempUser(createUserDto.email);
+
+    if (deleteTempUser) {
+      return this.userRepository.save(newUser);
+    } else {
+      throw new CustomRpcException(HttpStatus.BAD_REQUEST, '템프 유저 삭제 실패');
```

Comment:
외부에 표현해줄때는 요렇게가 더 명확하지 않을까 싶네요~
```suggestion
      throw new CustomRpcException(HttpStatus.BAD_REQUEST, '임시 유저 삭제 실패');
```

Reply:
[b11157c](https://github.com/GrebbMe/grabbme-be/pull/53/commits/b11157c7d78c36c8268cc7813c54f08f430ee33d)
클라이언트에 노출 되는 것 말씀해주신 문구가 더 와닿는것같습니다!
변경하여 반영하였습니다!